File: pyiArduinoI2Cencoder/examples/KeyPressed.py
#   ПРИМЕР РАБОТЫ С КНОПКОЙ ЭНКОДЕРА:
# В stdout выводится состояние кнопки
# энкодера на момент запуска сценария

# Подключаем библиотеку для работы с энкодером I2C-flash.
from pyiArduinoI2Cencoder import *
# Инстанцируем объект, указывая адрес модуля на шине I2C.
enc = pyiArduinoI2Cencoder(0x09)

# Считываем состояние кнопки энкодера в переменную f.
f = enc.getButton(KEY_PRESSED)

if not f:
    print("OFF")
else:
    print("ON")

File: pyiArduinoI2Cencoder/cpp/KeyPressed.cpp
/*	ПРИМЕР РАБОТЫ С КНОПКОЙ ЭНКОДЕРА:
 * В stdout выводится состояние кнопки в момент
 * запуска программы.
 */

#include <cstdio>
// Подключаем библиотеку для работы с энкодером I2C-flash.
#include "../iarduino_I2C_Encoder.h"

// Инстанциируем объект энкодера с адресом
iarduino_I2C_Encoder enc(0x09);

int main()
{
	// Инициируем работу с энкодером.
	enc.begin();
	// Считываем состояние кнопки энкодера в переменную f.
	bool f = enc.getButton(KEY_PRESSED);
	printf("%s", f ? "ON \n" : "OFF\n");
}
